expression/builtin_time.go

```diff
@@ -391,11 +390,11 @@ type builtinFromDaysSig struct {
 func (b *builtinFromDaysSig) eval(row []types.Datum) (d types.Datum, err error) {
 	args, err := b.evalArgs(row)
 	if err != nil {
-		return types.Datum{}, errors.Trace(err)
+		return d, errors.Trace(err)
 	}
 	days, err := args[0].ToInt64(b.ctx.GetSessionVars().StmtCtx)
 	d.SetMysqlTime(types.TimeFromDays(days))
-	return d, nil
+	return d, err
```

Comment:
If args[0] is a string like "z500z",
ToInt64 will cause a TruncatedError,
thus, [test case](https://github.com/pingcap/tidb/blob/master/expression/builtin_time_test.go#L817) can not be passed if we change `return d, nil` to `return d, err`.

You can keep it `return d, nil` currently.